app/models/concerns/hyku_addons/user_behavior.rb

```diff
@@ -4,30 +4,40 @@ module HykuAddons
   module UserBehavior
     extend ActiveSupport::Concern
 
+    PROFILE_VISIBILITY = { open: "open", closed: "closed" }.freeze
+
     included do
-      around_update :toggle_display_profile
-      validate :must_have_valid_email_format
-    end
+      before_save :toggle_display_profile
 
-    def must_have_valid_email_format
-      accepted_email_formats = current_account&.settings&.dig('email_format')
-      return unless accepted_email_formats.present?
+      validate :email_format
 
-      email_format = '@' + email.split('@')[-1]
-      errors.add(:email, "Email must contain #{accepted_email_formats.join(', ')}") unless accepted_email_formats.include? email_format
+      scope :with_public_profile, -> { where(display_profile: true) }
     end
 
+    # NOTE: Is this used anywhere?
```

Comment:
It looks like it was used on line 30, and has now been removed and replaced with `Site.account`.  But a search of the code base turns up results for this method so perhaps double check before you delete it.  But we might as well clear this up now.

Reply:
@BertZZ you added this method, did you just use it in this class, or was it used elsewhere too?